src/cargo_deny_list.rs

```diff
@@ -0,0 +1,246 @@
+//! Module for parsing the output of the calls to `cargo deny`.
+//!
+//! The assumption is that the input to the [CargoDenyList] type is the result of running the
+//! command `cargo deny list -l crate -f json`.
+//!
+//! The result of such a command is a JSON object with the following structure:
+//!
+//! ```json
+//! {
+//!    "adler2 2.0.0 registry+https://github.com/rust-lang/crates.io-index": {
+//!       "licenses": [
+//!         "0BSD",
+//!        "MIT",   
+//!       "Apache-2.0"
+//!      ]
+//!   },
+//!
+//! // More dependencies...
+//! }
+//! ```
+//!
+//! This module parses this JSON object into a more usable format for our use case.
+//! See [CargoDenyList] type for more information.
+use std::collections::HashMap;
+
+use serde::{Deserialize, Deserializer, Serialize};
+
+/// Represents the output of a call to `cargo deny list -l crate -f json`, adapted to our use case.
+///
+/// The output of that command is a JSON object. The keys are strings with the following pattern:
+///
+/// ```text
+/// <CRATE_NAME> <CRATE_VERSION> <REGISTRY_TYPE>+<URL>
+/// ```
+///
+/// While the values are JSON objects with the following structure:
+/// ```json
+/// {
+///   "licenses": [
+///     "MIT",
+///     "Apache-2.0"
+///   ]
+/// }
+///
+/// We are only interested in the dependency name, its URL, and the licenses it uses.
+/// So that's what we store.
+#[derive(Debug, Serialize, PartialEq)]
+pub(super) struct CargoDenyList {
+    /// Map of dependencies and their licenses.
+    dependencies: HashMap<NameAndUrl, LicenseList>,
+}
+
+impl<'de> Deserialize<'de> for CargoDenyList {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        Ok(CargoDenyList {
+            dependencies: HashMap::deserialize(deserializer)?,
+        })
+    }
+}
+
+/// Represents the licenses used by a dependency.
+///
+/// The `licenses` field is a list of strings with the short names
+/// of the licenses used by the dependency.
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+#[cfg_attr(test, derive(Default))]
+struct LicenseList {
+    /// List of licenses used by the dependency.
+    licenses: Vec<String>,
+}
+
+/// Represents the name of a dependency and its URL.
+///
+/// The format of this string is expected to be:
+///
+/// ```text
+/// <CRATE_NAME> <URL>
+/// ```
+///
+/// The URL is formatted as a link to the dependency's page.
+#[derive(Debug, PartialEq, Eq, Hash, Serialize)]
+struct NameAndUrl(String);
+
+impl<'de> Deserialize<'de> for NameAndUrl {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let key = String::deserialize(deserializer)?;
+        let key_split = key.split_whitespace().collect::<Vec<_>>();
+
+        match key_split[..] {
+            [crate_name, _crate_version, url] => Ok(NameAndUrl(format!(
```

Comment:
This drops the crate version from the input so it's not usable elsewhere in the code. Reduces complexity for the templates, but performs the changes via code logic so it's less flexible. Let me know what you prefer.